sdk/lib/crt/math/ceil.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * PROJECT:     ReactOS CRT library
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Portable implementation of ceil
+ * COPYRIGHT:   Copyright 2021 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+#include <limits.h>
+
+#ifdef _MSC_VER
+#pragma function(ceil)
+#endif
+
+double
+__cdecl
+ceil(double x)
+{
+    /* Load the value as uint64 */
+    unsigned long long u64 = *(unsigned long long*)&x;
+
+    /* Check for NAN */
+    if ((u64 & ~(1ULL << 63)) > 0x7FF0000000000000ull)
+    {
+        /* Set error bit */
+        u64 |= 0x0008000000000000ull;
+        return *(double*)&u64;
+    }
+
+    /* Check if x is positive */
+    if ((u64 & (1ULL << 63)) == 0)
+    {
+        /* Check if it fits into an int64 */
+        if (x < (double)_I64_MAX)
+        {
+            long long ll = (long long)x;
```

Comment:
Is it possible to have "ll" as a double, and doing the `(double)(long long)` trick on x? If so this would avoid the repetitive `(double)ll` casts below :thinking:

Reply:
Fixed